os/arch/arm/src/imxrt/imxrt_pit.c

```diff
@@ -151,3 +216,381 @@ uint64_t imxrt_pit_getlifetimetimercount(PIT_Type *base)
+ *              function pointer is NULL, then the reset-on-expiration
+ *              behavior is restored,
+ *  arg       - Argument that will be provided in the callback
+ *
+ * Returned Values:
+ *   The previous timer expiration function pointer or NULL is there was
+ *   no previous function pointer.
+ *
+ ****************************************************************************/
+static void imxrt_pit_setcallback(struct timer_lowerhalf_s *lower,
+								tccb_t callback, void *arg)
+{
+	struct imxrt_pit_lowerhalf_s *priv = (struct imxrt_pit_lowerhalf_s *)lower;
+	irqstate_t flags = irqsave();
+
+	/* Save the new callback */
+	priv->callback = callback;
+	priv->arg      = arg;
+
+	if (priv->callback != NULL) {
+		imxrt_pit_setisr(priv->pit, imxrt_pit_handler, priv);
+		imxrt_pit_enableinterrupts(priv->pit->base, priv->channel, kPIT_TimerInterruptEnable);
+	} else {
+		imxrt_pit_disableinterrupts(priv->pit->base, priv->channel, kPIT_TimerInterruptEnable);
+		imxrt_pit_setisr(priv->pit, NULL, NULL);
+	}
+
+	irqrestore(flags);
+}
+
+/****************************************************************************
+ * Name: imxrt_pit_ioctl
+ *
+ * Description:
+ *   Any ioctl commands that are not recognized by the "upper-half" driver
+ *   are forwarded to the lower half driver through this method.
+ *
+ * Input Parameters:
+ *   lower - A pointer the publicly visible representation of the "lower-half"
+ *           driver state structure.
+ *   cmd   - The ioctl command value
+ *   arg   - The optional argument that accompanies the 'cmd'.  The
+ *           interpretation of this argument depends on the particular
+ *           command.
+ *
+ * Returned Values:
+ *   Zero on success; a negated errno value on failure.
+ *
+ ****************************************************************************/
+static int imxrt_pit_ioctl(struct timer_lowerhalf_s *lower, int cmd,
+						unsigned long arg)
+{
+	return OK;
+}
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+int imxrt_pit_initialize(const char *devpath, int timer)
+{
+	pit_config_t config;
+	struct imxrt_pit_lowerhalf_s *lower = NULL;
+
+	switch (timer) {
+	case IMXRT_PIT_CH0:
+		lower = &g_pit0_lowerhalf;
+		break;
+
+	case IMXRT_PIT_CH1:
+		lower = &g_pit1_lowerhalf;
+		break;
+
+	case IMXRT_PIT_CH2:
+		lower = &g_pit2_lowerhalf;
+		break;
+
+	case IMXRT_PIT_CH3:
+		lower = &g_pit3_lowerhalf;
+		break;
+	}
+
+	if (!lower) {
+		return -ENODEV;
+	}
+
+	/* Initialize the elements of lower half state structure */
+	lower->started  = false;
+	lower->callback = NULL;
+
+	imxrt_pit_getdefaultconfig(&config);
+	memcpy(&(lower->config), &config, sizeof(pit_config_t));
+	lower->pit = imxrt_pit_configure(timer, &config, timer);
+	lower->channel = timer;
+
+	if (lower->pit == NULL) {
+		return -EINVAL;

```

Comment:
You didn't use any of parameter here, It's not -EINVAL.
If you add pit_config_t and use it here, then EINVAL can be accepted.

Reply:
Updated